tempesta_fw/connection.h

```diff
@@ -164,10 +164,9 @@ static inline bool
 tfw_connection_put(TfwConnection *conn)
 {
 	int rc;
-
-	if (unlikely(!conn))
-		return false;
-
+	rc = atomic_read(&conn->refcnt);
+	if (unlikely(!rc || rc == TFW_CONN_DEATHCNT))
+		return true;
 	rc = atomic_dec_return(&conn->refcnt);
 	if (unlikely(!rc || rc == TFW_CONN_DEATHCNT))
```

Comment:
With the latest changes, I don't think `refcnt` ever reaches zero. The last reference is held by the connection itself, and at that time `TFW_CONN_DEATHCNT` is set already.